### PR TITLE
fix(config): warn if cve is used with remote storage driver

### PR DESCRIPTION
**Pull request by chofnar**
**What type of PR is this?**

config hardening

**What does this PR do / Why do we need it**:

Checks if remote storage driver is used while CVE is enabled at config verification time, warns that they are incompatible and exits with BadConfig. Also adjusts tests to reflect this.

**If an issue # is not available please add repro steps and logs showing the issue**:

If running zot with cve enabled and an s3 storage driver, the cve extension will treat the s3 root directory as a local directory leading to unexpected behavior

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rchincha** [comment]
But what happens if I actually make a CVE query request?

**Comment by chofnar** [comment]
> But what happens if I actually make a CVE query request?

I tried with CVEListForImage in the GQL Playground after I added some images with skopeo and it seems to be working

File: pkg/cli/root.go [hunk 1]
```diff
@@ -306,6 +306,24 @@ func validateCacheConfig(cfg *config.Config) error {
 	return nil
 }
 
+func validateExtensionsConfig(cfg *config.Config) {
+	//nolint:lll
+	if cfg.Storage.StorageDriver != nil && cfg.Extensions != nil && cfg.Extensions.Search != nil && cfg.Extensions.Search.CVE != nil {
```
Comment (peusebiu) [inline pkg/cli/root.go:311]: do not modify the internal config :D, will create issues with the configloader.

just return an error, look at sync+s3 combination which also panics

Comment (chofnar) [reply pkg/cli/root.go:311]: Added

**Review by rchincha — DISMISSED** [review]
lgtm

File: examples/config-search.json [hunk 2]
```diff
@@ -12,10 +12,7 @@
     },
     "extensions": {
         "search": {
-          "enable": true,
```
Comment (rchincha) [inline examples/config-search.json:15]: This is not correct. Existing configurations should work as is. Add a new example if need be called config-remote-only.json

**Pull request closed without merging**